routes/anonymousDeck.js

```diff
@@ -108,6 +98,90 @@ router.post("/create", async function (req, res) {
   }
 });
 
+router.post("/edit", async function (req, res) {
+  try {
+    userId = await routeUtils.verifyLoggedIn(req);
+    let deckId = String(req.body.id);
+
+    let deck = await models.AnonymousDeck.findOne({
+      id: deckId,
+    })
+      .select("_id id name creator")
+      .populate("creator", "id");
+
+    if (!deck || deck.creator.id != userId) {
+      res.status(500);
+      res.send("You can only edit decks you have created.");
+      return;
+    }
+
+    deck.editing = Boolean(req.body.editing);
+    deck.id = String(req.body.id || "");
+    deck.name = String(req.body.name || "");
+
+    // deck name
+    if (!deck.name || !deck.name.length) {
+      res.status(500);
+      res.send("You must give your deck a name.");
+      return;
+    }
+    if (deck.name.length > constants.maxDeckNameLength) {
+      res.status(500);
+      res.send("Deck name is too long.");
+      return;
+    }
+
+    let profiles = await models.AnonymousDeck.findOne({ id: deckId }).select(
+      "profiles"
+    );
+
+    profiles = await models.DeckProfile.find({ _id: { $in: profiles } }).select(
+      "_id name avatar color deathMessage"
+    );
+
+    //Case 1: No profiles exist, so just create new ones.
```

Comment:
will this not be created in /profiles/create?

this /edit function looks like it's not doing anything?

and i only see a case#1 here